File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/config/impl/RConfig.java
package com.dkellycollins.triggerfinger.data.config.impl;

import android.content.Context;
import android.util.TypedValue;

public abstract class RConfig {

    protected final Context _context;

    protected RConfig(Context context) {
        _context = context;
    }

    protected float getFloat(int id) {
        TypedValue outValue = new TypedValue();
        _context.getResources().getValue(id, outValue, true);

        return outValue.getFloat();
    }

}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/config/impl/RConfig.kt
package com.dkellycollins.triggerfinger.data.config.impl

import android.content.Context
import android.util.TypedValue

abstract class RConfig protected constructor(protected val _context: Context) {

    protected fun getFloat(id: Int): Float {
        val outValue = TypedValue()
        _context.resources.getValue(id, outValue, true)

        return outValue.float
    }

}
